src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1256,7 +1261,13 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
                 }
                 try
                 {
-                    bodyValue = await httpContext.Request.ReadFromJsonAsync(bodyType, jsonSerializerOptions);
+                    // Edge case but possible if the RequestDelegateFactoryOptions.ServiceProvider and
+                    // RequestDelegateFactoryOptions.EndpointBuilder.ServiceProvider are null
+                    // In this situation both options and jsonTypeInfo are null.
+                    jsonSerializerOptions ??= httpContext.RequestServices.GetService<IOptions<JsonOptions>>()?.Value.SerializerOptions ?? JsonOptions.DefaultSerializerOptions;
+                    jsonTypeInfo ??= jsonSerializerOptions.GetTypeInfo(bodyType);
```

Comment:
Why do we do this? Don't we already have an instance of the options on the factory context (https://github.com/dotnet/aspnetcore/blob/8f6b7ab4bfe1cb80666747942c07249158b64257/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L279)

Reply:
I don't totally agree with that but there is a possibility to have a `null` `JsonSerializerOptions` in the `FactoryContext` that was decided to not set to the default https://github.com/dotnet/aspnetcore/pull/45359#discussion_r1036632987. I don't want to change it, so, I'm fall backing to get from DI container at request time.

Reply:
> we shouldn't have any per request fallback.

💯 and I can change it but this fallback is not been introduced in this pr. @BrennanConroy do you remember the reason why you decided to not use the default options when not in the DI?

Reply:
Idk, probably because it's a super edge case that I didn't think was worth pushing a change and respinning the CI for.